source_manager.go

```diff
@@ -141,15 +158,121 @@ func NewSourceManager(an ProjectAnalyzer, cachedir string) (*SourceMgr, error) {
 		}
 	}
 
-	return &SourceMgr{
+	sm := &SourceMgr{
 		cachedir: cachedir,
 		lf:       fi,
 		srcs:     make(map[string]source),
 		srcfuts:  make(map[string]*unifiedFuture),
 		an:       an,
 		dxt:      pathDeducerTrie(),
 		rootxt:   newProjectRootTrie(),
-	}, nil
+		qch:      make(chan struct{}),
+	}
+
+	return sm, nil
+}
+
+// SetUpSigHandling sets up typical signal handling for a SourceMgr. It will
+// register a signal handler to be notified on:
+//
+//  - syscall.SIGINT
+//  - syscall.SIGHUP
+//  - syscall.SIGTERM
+//  - syscall.SIGQUIT
+//  - os.Interrupt
+func SetUpSigHandling(sm *SourceMgr) {
+	sigch := make(chan os.Signal)
+	sm.HandleSignals(sigch)
+	signal.Notify(sigch, syscall.SIGINT, syscall.SIGHUP, syscall.SIGTERM, syscall.SIGQUIT, os.Interrupt)
```

Comment:
`os.Interrupt` is a `syscall.SIGINT` so you can drop the `syscall.SIGINT` here. As for HUP/TERM/QUIT unless you're doing something different with them, I wouldn't bother handling them, Interrupt is enough and very cross platform.